drivers/libusb1.c

```diff
@@ -1056,7 +975,7 @@ static int nut_libusb_set_report(
 
 /* Expected evaluated types for the API:
  * static int nut_libusb_get_string(libusb_device_handle *udev,
- *	int StringIdx, char *buf, int buflen)
+ *	uint8_t StringIdx, unsigned char *buf, uint16_t buflen)
```

Comment:
Hi, thanks for your review! Do you review commit-per-commit or only the final diff? It might be easier to follow if going through the commits.

I should point out here that the comment changes in "Fix up comments about expected types" update the comments to reflect the status quo (same before or after my changes, I didn't change anything in this regard). So this commit is technically independent of the other commits. It just makes the following commit easier to digest. Having these corrected comments in front of us, it is easier to see why the range-checks were flawed (and they are not gone, only reduced to what is needed, and at the same time removing the need for all the ugly pragmas).

In libusb1.c, StrIdx is uint8_t, so some code cannot pass 256 to it.

In libusb0.c, StrIdx is int, so for positive values, it cannot be larger than MAX_INT anyway. Maybe the original reason for some compiler warnings that were hidden with pragmas. However since it was cast to uintmax_t, an original negative value could indeed become higher than MAX_INT. But in that case the first < 0 would have caught that, so it is superfluous anyway. So the compiler would still be right to complain. Finally this all is useless anyway, because the highest usable value is 255.

I believe the original intention of the casting and comparing was to make something that would work for libusb0 and libusb1 with identical code. And not thinking about the practical limits of the value.

Reply:
For this review, I went with the final diff as attached to the PR. Reviewing individual in-flight commits proved problematic earlier, as force-pushes detached them from PR and project history.

As for the int type analysis - thanks, I concur. As for pragmas - good riddance :D

Reply:
Ah, you're reviewing per e-mail. In the web interface it is easy to go through each commit. Does it mean you haven't read the commit messages at all?

Reply:
I did read the messages... now :D (thanks for diligently filling those out). But alas not on the first pass by phone from the road.